core/src/main/java/org/frankframework/senders/ReconnectSenderWrapper.java

```diff
@@ -65,20 +65,36 @@ public void open() throws SenderException {
 			sender.open();
 			super.open();
 		} finally {
+			super.close();
 			sender.close(); //Only open to test the connection, close afterwards if all is ok.
 		}
 	}
 
 	@Override
 	public SenderResult doSendMessage(Message message, PipeLineSession session) throws SenderException, TimeoutException {
-		try {
-			sender.open();
-			return sender.sendMessage(message, session);
-		} finally {
-			sender.close();
-		}
+		sender.open();
+		session.scheduleCloseOnSessionExit(new AutoCloseableSenderWrapper(sender), ReconnectSenderWrapper.class.getSimpleName());
```

Comment:
```suggestion
		session.scheduleCloseOnSessionExit(new AutoCloseableSenderWrapper(sender), this);
```
The sole purpose of the requester argument is for traceability in the logging. It might be best to just print the entire object (which includes the hashCode).

Reply:
This suggestion does not compile, since the method only accepts a String. I'll fix it.